### DISCUSSION TITLE
Cloudflare started throwing Invalid SSL Certificate Error, Domain in vercel stuck on Generating SSL Certificate

### DISCUSSION THREAD

**Discussion by mustafahasankhan** [vercel/community#5917] [Help]
### Summary

Folks,
All of a sudden my site that uses Cloudflare as reverse proxy + dns and deployed on Vercel started giving Error Code 526 Invalid SSL Certificate.

Under Domains in Settings of my Project on Vercel I see it is stuck on: Generating SSL Certificate
All other config on Cloudflare remains unchanged.

My SSL settings on Cloudflare is set to Full (strict), my domain's A record to Vercel's IP and www CNAME record to: cname.vercel-dns.com

I have tried removing and adding it, nothing has helped so far.

### Steps to Reproduce

Add a site on Cloudflare, set A record to Vercel's IP and CNAME record to cname.vercel-dns.com
Add your domain under Domains in Settings of your project.
It remains stuck on Generating SSL Certificate

**Comment by amyegan** [comment]
Is the domain working now, or do you still have this problem?

**Comment by BodieLeonard** [comment]
We are seeing the same SSL issue for international traffic. Getting error "Your connection to this sites not secure".

**Comment by amyegan** [comment]
Do you have Cloudflare Proxy enabled? 

https://vercel.com/docs/integrations/cloudflare

**Reply by BodieLeonard** [reply]
I do not. The DNS is managed by godaddy. Do you suggest I add Cloudflare to the mix? I am not sure that is required since the DNS is on godaddy and the SSL is managed on vercel.

**Reply by BodieLeonard** [reply]
I already had the DNS A record for the Vercel IP (76.76.21.21) and today added the CAA record via this post https://vercel.com/guides/a-record-and-caa-with-vercel

**Reply by amyegan** [reply]
No need to add Cloudflare to the mix. I asked because the original poster mentioned using it, but I wasn't sure which settings they used.

How long has it been since you added the CAA record?

**Reply by BodieLeonard** [reply]
It has been a few hours since I added the CAA
